src/number_theory/divisors.lean

```diff
@@ -398,4 +403,82 @@ begin
   exact nat.lt_succ_of_le (nat.divisor_le (nat.mem_divisors.2 ⟨ha, h⟩))
 end
 
+
+
+-- # Prime divisors
+
+lemma prime_divisors_mem {n p : ℕ} (hn : n ≠ 0) : p ∈ n.prime_divisors ↔ p ∣ n ∧ prime p :=
+by { unfold prime_divisors, rw [mem_filter, mem_divisors],
+    rw [and.congr_left_iff, and_iff_left_iff_imp], intros, exact hn }
+
+/-- 0 has no prime divisors -/
+lemma prime_divisors_zero : prime_divisors 0  = (∅ : finset ℕ) :=
+by { unfold prime_divisors divisors, simp }
+
+/-- 1 has no prime divisors -/
+lemma prime_divisors_one : prime_divisors 1 = (∅ : finset ℕ) :=
+by { apply filter_singleton prime }
+
+lemma prime_divisors_eq_to_finset_factors (n : ℕ) : n.prime_divisors = n.factors.to_finset :=
+begin
+  by_cases hn : n = 0,
+  { rw [hn, prime_divisors_zero], simp },
+  { ext,
+    rw [prime_divisors_mem hn, list.mem_to_finset, mem_factors (pos_iff_ne_zero.mpr hn), and_comm] }
+end
+
+/-- The sets of prime divisors of coprime `a` and `b` are disjoint -/
+lemma coprime_prime_divisors_disjoint {a b : ℕ} (h_ab_coprime : coprime a b) :
+  disjoint a.prime_divisors b.prime_divisors :=
+begin
+  by_cases ha : a = 0, { rw ha at *, rw prime_divisors_zero, apply disjoint_empty_left },
+  by_cases hb : b = 0, { rw hb at *, rw prime_divisors_zero, apply disjoint_empty_right },
+
+  rw disjoint_iff_ne,
+  intros x hx y hy,
+  rw (prime_divisors_mem ha) at hx,
+  rw (prime_divisors_mem hb) at hy,
+  cases hy with hy1 hy2,
+  by_contra H, rw H at hx,
+  suffices : y = 1, { rw this at hy2, exact not_prime_one hy2 },
+  exact eq_one_of_dvd_coprimes h_ab_coprime hx.1 hy1,
+end
+
+/-- For coprime `a` and `b`, prime divisors of `(a * b)` are the union of those of `a` and `b` -/
+lemma coprime_prime_divisors_product {a b : ℕ} (h_ab_coprime : coprime a b) :
+  (a * b).prime_divisors = a.prime_divisors ∪ b.prime_divisors :=
+begin
+  by_cases ha0 : a = 0,
+    { rw [ha0, coprime_zero_left] at h_ab_coprime,
+      rw [ha0, h_ab_coprime, zero_mul, prime_divisors_zero, prime_divisors_one], simp },
+  by_cases hb0 : b = 0,
+    { rw [hb0, coprime_zero_right] at h_ab_coprime,
+      rw [hb0, h_ab_coprime, mul_zero, prime_divisors_zero, prime_divisors_one], simp },
+  have hab : a * b ≠ 0, { exact mul_ne_zero ha0 hb0 },
```

Comment:
You should extract everything after this point to a standalone `prime_divisors_mul_of_ne_zero` lemma.